src/openforms/payments/contrib/worldline/plugin.py

```diff
@@ -246,30 +242,22 @@ def handle_return(
 
             checkout_status = WorldlineHostedCheckoutStatus(response.status)
             payment_data = response.created_payment_output
-            try:
+
+            if payment_data and payment_data.payment:
                 status = WorldlinePaymentStatus(
                     payment_data.payment.status
                     if payment_data and payment_data.payment
                     else None
                 )
-            except ValueError:
+
+                payment.provider_payment_id = payment_data.payment.id
+                payment.save(update_fields=("provider_payment_id",))
+            else:
                 status = WorldlineHostedCheckoutStatus.to_payment_status(
                     checkout_status
```

Comment:
If we want to adhere to the minimum amount of coverage we could chose to remove these lines (as the status also could be updated at a later point through the webhooks) or we could create a test for this with mocks 🤔